### PR TITLE
Fix kie-issues #1236 - Replace quay.io with docker.io, review version.go

### PR DESCRIPTION
**Pull request by ricardozanini**
**Description of the change:**
Fixes https://github.com/apache/incubator-kie-issues/issues/1236

**Motivation for the change:**
We need to realign the images with docker.io in order to release under Apache.

**Checklist**

- [ ] Add or Modify a unit test for your change
- [ ] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by ricardozanini** [comment]
@tiagobento @thiagoelg fyi

**Comment by ricardozanini** [comment]
@jstastny-cz I think we need to align the Jenkins CI part, since the weekly I guess we will publish the `main` tag, right? And the release, the `operatorVersion`.

File: container-builder/examples/dockerfiles/SonataFlow.dockerfile
Comment (thiagoelg) [inline container-builder/examples/dockerfiles/SonataFlow.dockerfile:18]:
```suggestion
FROM docker.io/apache/incubator-kie-sonataflow-builder:main AS builder
```

**Review by thiagoelg — CHANGES_REQUESTED** [review]
Some small typos

**Comment by ricardozanini** [comment]
@thiagoelg please remove the request for changes

e2e is failing duo to the images renaming

**Comment by thiagoelg** [comment]
> @thiagoelg please remove the request for changes
> 
> e2e is failing duo to the images renaming

It seems that it's failing with a similar error we got on kie-tools, this change might fix it: https://github.com/apache/incubator-kie-tools/pull/2347

**Comment by ricardozanini** [comment]
@thiagoelg thanks! It worked: https://github.com/apache/incubator-kie-kogito-serverless-operator/pull/469

I'm gonna rebase it.

**Review by wmedvede — APPROVED** [review]

**Review by jstastny-cz — APPROVED** [review]
LGTM.

**Comment by rodrigonull** [comment]
@ricardozanini can we merge this PR? We need to merge this one and the https://github.com/apache/incubator-kie-kogito-serverless-operator/pull/494 to unblock the seed jobs on Apache Jenkins

**Comment by ricardozanini** [comment]
@rodrigonull sure, we just need @thiagoelg to remove the "require changes" review. The merge is blocked until there.

**Review by thiagoelg — APPROVED** [review]

**Pull request merged**
